tests/ui/admin_flow_test.py

```diff
@@ -0,0 +1,155 @@
+"""Test admin flows, for now just add."""
+import unittest
+
+import flask
+import json
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from base_test import BaseTest
+from login_page import LoginPage
+from layout import UfOPageLayout
+
+
+class AdminFlowTest(BaseTest):
+
+  """Test admin flows."""
+  TEST_ADMIN_AS_DICT = {
+      'username': 'Test Admin Fake Name 01',
+      'password': 'fake admin password that could be anything'
+  }
+
+  def setUp(self):
+    """Setup for test methods."""
+    super(AdminFlowTest, self).setUp()
+    super(AdminFlowTest, self).setContext()
+
+  def tearDown(self):
+    """Teardown for test methods."""
+    self.removeTestAdmin(should_raise_exception=False)
+    super(AdminFlowTest, self).tearDown()
+
+  def testAddingAnotherAdminWorks(self):
+    """Test that adding another admin works and lets us login as that admin."""
+    # Try to login with the uncreated admin to ensure it does not exist.
+    LoginPage(self.driver).Login(self.args.server_url,
+                                 self.TEST_ADMIN_AS_DICT['username'],
+                                 self.TEST_ADMIN_AS_DICT['password'])
+    self.driver.get(self.args.server_url + flask.url_for('landing'))
+
+    # Assert that login failed (we're still on the login page).
+    login_url = self.args.server_url + flask.url_for('login')
+    self.assertEquals(login_url, self.driver.current_url)
+
+    # Login as an existing admin to get access.
+    LoginPage(self.driver).Login(self.args.server_url, self.args.username,
+                                 self.args.password)
+    self.driver.get(self.args.server_url + flask.url_for('landing'))
+
+    # Find the add admin form.
+    add_admin_form = self.getAdminFormHelper()
+    response_status = add_admin_form.find_element(
+        *UfOPageLayout.ADD_ADMIN_RESPONSE_STATUS)
+    self.assertIsNone(response_status)
+
+    # Add the new admin.
+    username_input = add_admin_form.find_element(
+        *UfOPageLayout.ADD_ADMIN_USERNAME)
+    username_real_input = username_input.find_element(By.ID, 'input')
```

Comment:
nit: hmmm, I think the "username_real_input" here should actually be "username_input".  And above, maybe "username_input" should better be "username_row" (or something that describes what it is)????